src/host/ft_fuzzer/fuzzmain.cpp

```diff
@@ -0,0 +1,155 @@
+
+    // it is safe to pass INVALID_HANDLE_VALUE here because the null handle would have been detected
+    // in ConDrvDeviceComm (which has been avoided by setting a global device comm beforehand)
+    RETURN_IF_NTSTATUS_FAILED(ConsoleCreateIoThreadLegacy(INVALID_HANDLE_VALUE, args));
+
+    auto& gci = Microsoft::Console::Interactivity::ServiceLocator::LocateGlobals().getConsoleInformation();
+
+    // Process handle list manipulation must be done under lock
+    gci.LockConsole();
+    ConsoleProcessHandle* pProcessHandle{ nullptr };
+    RETURN_IF_FAILED(gci.ProcessHandleList.AllocProcessData(GetCurrentProcessId(),
+                                                            GetCurrentThreadId(),
+                                                            0,
+                                                            nullptr,
+                                                            &pProcessHandle));
+    pProcessHandle->fRootProcess = true;
+    gci.UnlockConsole();
+
+    constexpr static std::wstring_view fakeTitle{ L"Fuzzing Harness" };
+
+    CONSOLE_API_CONNECTINFO fakeConnectInfo{};
+    fakeConnectInfo.ConsoleInfo.SetShowWindow(SW_NORMAL);
+    fakeConnectInfo.ConsoleInfo.SetScreenBufferSize(til::size{ 80, 25 });
+    fakeConnectInfo.ConsoleInfo.SetWindowSize(til::size{ 80, 25 });
+    fakeConnectInfo.ConsoleInfo.SetStartupFlags(STARTF_USECOUNTCHARS);
+    wcscpy_s(fakeConnectInfo.Title, fakeTitle.data());
+    fakeConnectInfo.TitleLength = gsl::narrow_cast<DWORD>(fakeTitle.size() * sizeof(wchar_t)); // bytes, not wchars
+    wcscpy_s(fakeConnectInfo.AppName, fakeTitle.data());
+    fakeConnectInfo.AppNameLength = gsl::narrow_cast<DWORD>(fakeTitle.size() * sizeof(wchar_t)); // bytes, not wchars
+    fakeConnectInfo.ConsoleApp = TRUE;
+    fakeConnectInfo.WindowVisible = TRUE;
+    RETURN_IF_NTSTATUS_FAILED(ConsoleAllocateConsole(&fakeConnectInfo));
+
+    CommandHistory::s_Allocate(fakeTitle, (HANDLE)pProcessHandle);
+
+    gci.UnlockConsole(); // Locked in CAC
+
+    return S_OK;
+}
+
+extern "C" __declspec(dllexport) HRESULT RunConhost()
+{
+    Microsoft::Console::Interactivity::ServiceLocator::LocateGlobals().hInstance = wil::GetModuleInstanceHandle();
+
+    // passing stdin/stdout lets us drive this like conpty (!!) and test the VT renderer (!!)
+    // but for now we want to drive it like conhost
+    ConsoleArguments args({}, nullptr, nullptr);
+
+    HRESULT hr = args.ParseCommandline();
+    if (SUCCEEDED(hr))
+    {
+        hr = StartNullConsole(&args);
+    }
+
+    return hr;
+}
+
+#ifdef FUZZING_BUILD
+extern "C" __declspec(dllexport) int LLVMFuzzerInitialize(int* /*argc*/, char*** /*argv*/)
+#else
+int main(int /*argc*/, char** /*argv*/)
+#endif
+{
+    RETURN_IF_FAILED(RunConhost());
+    return 0;
+}
+
+extern "C" __declspec(dllexport) int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
+{
+    if (size < 1)
+    {
+        return 0;
+    }

```

Comment:
I'd prefer if you let the fuzzer give emptyness here as that's an interesting case.